### ISSUE TITLE
Bloated Address Book

### ISSUE THREAD

**Issue by x80486** [signalapp/Signal-Android#10891]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description

Removed contacts from Address Book (`Contacts`) application are still on Signal — if contact was accepted to call/message before. So every time somebody changes the phone number and forgets to delete the Signal account, I get another entry in Signal and the Address Book (`Contacts`) application.

Basically, there is no way to remove a Signal contact.

### Steps to Reproduce

- Have someone (new) to call/message you from Signal
- Add contact details to Android's address book (`Contacts`) application
- Remove it at some point

**Actual result:** Contact number remains on the Address Book (as Signal call/message)
**Expected result:** Contact should be removed — not sure if after some period of inactivity, etc., but they keep piling up there

### Device Info

**Device:** Pixel 4a (5G)
**Android version:** 11 (Jan 5, 2021 update)
**Signal version:** 5.3.7.1

**Comment by a-d-collins** [comment]
I get the sense that you're implying that the "actual result" is a bug. I'm not so sure that that's the right idea. Contacts management, for example, is also not a concept in WhatsApp, so what your seeing is probably by design.

From my point of view, if I deleted an android contact and then that contact sent me a Signal message, it would be nice to know who it was from. That way, I can respond (or not respond) more informatively. I can even block that Signal user without having to interact, if that's what's deemed necessary.

My overall point is that maybe this issue could be redirected into a new _feature_ that allows a user to "hide" a Signal contact? @x80486 thoughts on that?

**Comment by x80486** [comment]
It looks like Signal maintains its own list of contacts, sometimes synchronized (or at least initially) with the phone's list of contacts.

After a phone number is added/modified/removed for a contact in the phone (and synchronized back with Signal, provided that person is using Signal, of course) it doesn't really reflect that change as someone would expect.

These are the two (annoying) scenarios that happens to me most of the time:

- An Android contact joins Signal but we haven't established any communication whatsoever at this point via Signal — I just got the notification from joining. I remove it from the Android contacts not knowing if they did something on their end — that contact details stays in Signal forever
- An Android contact joins Signal. We establish communication (message/call); the person changes the phone number and forgets to remove the account with the previous number...now you have two of them (because she/he joined with the new number), even though in the Android contacts the old number was removed

I consider that a bug. It could be rephrased (as anything in this life), but in the end, I think something is needed to "unfriend" that contact you once established communication with...and on top of that it looks like the synchronization doesn't quite works, IMO.

The reason I don't like _to block_ that phone number is because I'm only removing that phone number from my list; it could go to anyone else I don't know (more likely), but it could go to someone else I might know. If blocking a number puts it in that initial state where everyone is at when the first communication is about to be established, then it's fine, but if by blocking means that the phone number is not able to communicate with you anymore, I wouldn't like that.

What I'm trying to say is that there could be an action that puts a phone number in "state zero", and you have to approve that number in order to establish communication again, be it the same person or someone else (more likely) the owner.

**Comment by a-d-collins** [comment]
@x80486 have you closed and restarted the Signal app? The Signal contact no longer appears in Signal if I:

1. Remove the contact from android contacts
2. Close Signal
3. Reopen Signal

**Comment by x80486** [comment]
Indeed. All of that and also tried to re-sync (all the times) in between, but the contacts are dangling there in Signal.

**Comment by nkr0** [comment]
Same here. Signal has it's own address management system, and I think that's part of signal's future plans to have usernames. Still, there has to be a way to remove a contact.

**Comment by kosaimgmg2021** [comment]
Unsubscribe

၂၀၂၁၊ မတ် ၂၆၊ သော ၂၀:၃၂ တွင် nkr0 ***@***.***> က ရေးခဲ့သည်−

**Comment by miku86** [comment]
Any news?

I once added an anonymous number to my android contacts ("Aaa"), because I wanted to know if this person shares some details (photo, status) on WhatsApp or Signal.

Alright, after my research, I deleted them from my android contacts.
Now they are still in my Signal contacts and I can't delete them.

If I re-save them to my android contacts with a different name ("Zzz"), they show up with this new name ("Zzz") in Signal. If I then delete them again, they show up again with their old name ("Aaa").

**Comment by miku86** [comment]
Any news?
Or at least some valuable information if this is intended behavior or a bug?

Currently, I have 20 people in my android contact list, but 25 show up in the signal contact list. I tried every method I've found so far, but I can't get rid of them. These are old colleagues, so I don't want to see them in my contacts, but I also don't want to block them.

I even migrated to another smartphone with an empty contact list, but I always see these old contacts.

**Comment by julianfairfax** [comment]
I wrote a forum post that addresses this issue. If you're a user with feedback or experiences to share, or a developer who can actually take a look at implementing this, please check out the post: https://community.signalusers.org/t/handle-contacts-properly/36807

**Comment by miku86** [comment]
@julianfairfax 

Thanks for the great write-up.

**Comment by miku86** [comment]
Yes, it is still relevant.

**Comment by ZwergNaseXXL** [comment]
Not sure if I should open a new issue for this. I have never tried removing contacts from signal. My problem is rather that for every new Signal contact, the previously existing Android contact is unnecessarily duplicated in the Android address book. That's the real bloat.

**Comment by est61** [comment]
I too have the same issue, and this has been ongoing for a **very** long time.

The mere fact that Signal stores your contacts on their servers is scary enough already, the actual fact that they keep deleted contacts even though you have deleted them off your phone, is not even more scary but very annoying.
Here https://github.com/signalapp/Signal-Android/issues/11633#issuecomment-923241107 you say that in order to get a few contacts "truly" deleted is by deleting our whole Signal app account, and another person here says that that is probably "by design"?  Which stupid person thought of this design? Unless Signal has a different agenda?

If it is indeed by design(since no-one here cares about this issue that has been opened more than a year ago) you mean you enjoy keeping thousands of records of users' phone numbers in your databases, then you are no different from whatsapp or any other commercial messenger.

When i delete a contact off my phone, i expect it to be deleted period!  Signal or any other app has no right to keep any copy of contacts, let alone the ones the user deleted.
It makes you wonder there is only ONE reason why messenger apps these days require your phone number to register, and that's to be evidently tracked. There is absolutely **no** reason to not have Signal use a username system rather than a phone number system to register unless Signal's agenda falls in someone else's hands.

This issue should be renamed to: `Signal keeps your contacts even though you've deleted them`

**Comment by greyson-signal** [comment]
If you delete someone you've never talked to from your system contacts, they will no longer show up in your signal contacts list. If you've talked with them, they will show up. They sort of need to be findable at that point. At minimum, you have a signal session with them, so there will always be some record of them in our local data store because they could contact you via that session in the future. Or someone else could buy their phone number and try to contact you, and we'd need to show a safety number change alert. Does that make sense?

I do think one day we will have a better UI for managing contacts so that you could hide certain ones from your contact list, or maybe even delete all record of them. But this is very much a case of where you think things are simple, but in reality they are more nuanced.

> you enjoy keeping millions of records of users' phone numbers in your databases, then you are no different from whatsapp or any other commercial messenger.

This is like saying that because messages go through our servers, we are no different from any other commercial messenger. This would also be a silly statement, because messages, just like our server-side storage, are encrypted with keys only the user has access to and is unreadable by the Signal service. It's a pretty big difference. And if you still don't want your contacts to be restorable, you can [disable Signal PINs](https://support.signal.org/hc/en-us/articles/360007059792-Signal-PIN#pin_disable).

I'm closing this issue because I consider the desire (a full-featured in-app contact management system) to be a feature request, and we only track bugs on github. Thanks!

**Comment by miku86** [comment]
Just for people who are interested in the feature request: https://community.signalusers.org/t/allow-deletion-of-contacts/

Please upvote there.
